### PR TITLE
DPI scaling in GTK

### LINKED ISSUE #7766: [Skia.GTK] DPI scaling
## What would you like to be added:

DPI Scaling support on GTK

## Why is this needed:

Support for HI-DPI screens

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [x] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #7766

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## What is the current behavior?

Not scaling!

## What is the new behavior?

Support for DPI scaling on GTK.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Runtime.Skia.Gtk/Helpers/Dpi/DpiHelper.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Gtk;
+using Uno.Helpers;
+using Uno.UI.Runtime.Skia.Helpers.Windows;
+
+namespace Uno.UI.Runtime.Skia.Helpers.Dpi
+{
+	public class DpiHelper
```
Comment (jeromelaban) [inline src/Uno.UI.Runtime.Skia.Gtk/Helpers/Dpi/DpiHelper.cs:9]: Likely should be private.

File: src/Uno.UI.Runtime.Skia.Gtk/Helpers/Dpi/WindowsDpiUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		int ReloadSystemMetrics();
+
+		[PreserveSig]
+		void GetDesktopDpi(out float dpiX, out float dpiY);
+
+		// the rest is not implemented as we don't need it
+	}
+}
+
+public enum MonitorDpiType
```
Comment (jeromelaban) [inline src/Uno.UI.Runtime.Skia.Gtk/Helpers/Dpi/WindowsDpiUtilities.cs:114]: Place in a different file

**Review by jeromelaban — APPROVED** [review]
Make sure to add `#nullable enable` in all new files.

File: src/Uno.UI.Runtime.Skia.Gtk/Helpers/Dpi/DpiHelper.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+				_dpi = dpi;
+				_dpiChangedWrapper.Event?.Invoke(this, EventArgs.Empty);
+			}
+		}
+
+		private float GetNativeDpi()
+		{
+			if (_window.Window == null)
+			{
+				return 96.0f; // GDK Window not initialized yet, default to 96 DPI.
```
Comment (ajpinedam) [inline src/Uno.UI.Runtime.Skia.Gtk/Helpers/Dpi/DpiHelper.cs:73]: It feels this value (96.0f) could be moved to a constant. We are using it in different places and different files. What do you think?

**Review by ajpinedam — APPROVED** [review]
Looks good!

**Pull request merged**